src/zenml/orchestrators/input_utils.py

```diff
@@ -64,7 +64,12 @@ def resolve_step_inputs(
             )
 
         try:
-            artifact = step_run.outputs[input_.output_name]
+            if len(step_run.outputs[input_.output_name]) > 1:
```

Comment:
This is the exact case the we talked about this morning in the design meeting I think.

I think we should be able to figure out which of the versions is the output that was returned by the step, and use that here instead of failing.

This is only possible if the following code does not work:
```python
@step
def my_step() -> Tuple[Annotated[int, "shared_name"], Annotated[int, "shared_name"]]:
  ...
```

Reply:
Adjusted, so it works for cases like
```python
from zenml import step, pipeline, save_artifact
from typing_extensions import Annotated
from zenml.client import Client

from zenml.models.v2.core.pipeline_run import PipelineRunResponse


@step(enable_cache=False)
def producer_step() -> (
    Annotated[int, "shared_name"]
):
    save_artifact(41, "shared_name")
    return 42

@step(enable_cache=False)
def consumer_step(shared_name: int, expected: int):
    assert shared_name == expected

@pipeline
def my_pipeline():
    a = producer_step()
    consumer_step(a,42)
    consumer_step(Client().get_artifact_version("shared_name",1),41, after=["producer_step"])


if __name__ == "__main__":
    try:
        Client().delete_artifact("shared_name")
    except KeyError:
        pass
    prr: PipelineRunResponse = my_pipeline()
    assert set([a.load() for a in prr.steps["producer_step"].outputs["shared_name"]]) == {41, 42}
```